src/main/java/TaskList.java

```diff
@@ -0,0 +1,188 @@
+import java.io.*;
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private String filePath = "./data/tasks.txt";
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    public Task getTask(int taskIndex) {
+        return tasks.get(taskIndex);
+    }
+
+    public int getSize() {
+        return tasks.size();
+    }
+
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    public void deleteTask(int taskIndex) throws BuddyException {
+        if (tasks.size() == 0 ) {
+            System.out.println("There are no tasks in your list:");
```

Comment:
I think instead of printing the error statement, you can throw a DukeException with the error message